### PR TITLE
Component docs refactor

### PR DESCRIPTION
**Pull request by benjgil**
## Proposed Changes

- split out the various core react docs template sections to individual components
- expose wrappers for building new templates layer by layer
- improve typing so that all the above are easy to consume and compose new docs templates from
- fix useFetchDocs typing (properties array, rather than single property) and move component to more sensible location
- includes merged code from https://github.com/teambit/bit/pull/5790 (anything to do with composition-app - should be merged asap)
- Tested on a separate repo to make sure docs render correctly and can now be customised. Working well

### CODE REVIEW

File: scopes/react/ui/compositions-app/composition-app.docs.mdx
Comment (GiladShoham) [inline scopes/react/ui/compositions-app/composition-app.docs.mdx:1]: why don't you have here the metadata section with labels and everything?

File: scopes/react/ui/docs-content/docs-content.tsx
Comment (KutnerUri) [inline scopes/react/ui/docs-content/docs-content.tsx:34]: prettify

File: scopes/react/ui/docs-content/docs-content.tsx
Comment (KutnerUri) [inline scopes/react/ui/docs-content/docs-content.tsx:21]:
```suggestion
```

File: scopes/react/ui/properties/properties.tsx
Comment (KutnerUri) [inline scopes/react/ui/properties/properties.tsx:7]: plz rename PropertiesParams -> PropertiesProps

File: scopes/react/ui/properties/properties.tsx
Comment (KutnerUri) [inline scopes/react/ui/properties/properties.tsx:12]: moving the fetch here could potentially really improve loading times.

however, what if we want to use docs data in other places?  (ie in CompositionsSummery, etc)

File: .bitmap
Comment (KutnerUri) [inline .bitmap:1521]: maybe remame to properties table?

File: .bitmap
Comment (KutnerUri) [inline .bitmap:1161]: I think you should rename to `composition-carousel` or `compositions-gallery`

File: .bitmap
Comment (KutnerUri) [inline .bitmap:1023]: I think we should group these new components under a `ui/preview` or `ui/docs` namespace

**Review by KutnerUri — APPROVED** [review]

File: scopes/react/ui/apply-providers/apply-providers.tsx
Comment (benjgil) [reply scopes/react/ui/apply-providers/apply-providers.tsx:25]: done

File: scopes/react/ui/docs-app/docs-app.module.scss
Comment (benjgil) [reply scopes/react/ui/docs-app/docs-app.module.scss:12]: this has just been moved from where it was previously. Not in the scope of this PR to change/update

File: scopes/react/ui/docs-app/docs-app.tsx
Comment (benjgil) [reply scopes/react/ui/docs-app/docs-app.tsx:14]: interesting, was just following the syntax in these components :) 
happy to change it

File: scopes/react/ui/docs-app/docs-app.tsx
Comment (benjgil) [reply scopes/react/ui/docs-app/docs-app.tsx:17]: this is just what was there. I dont want to get too deep into refactors if i'm not sure how it'll affect things downstream, not in the scope of this PR

File: scopes/react/ui/docs-app/docs-app.tsx
Comment (benjgil) [reply scopes/react/ui/docs-app/docs-app.tsx:37]: Again, just moving code around. I dont really want to hold up this big change to refactor all the code. This is to support a legacy feature that will likely be removed anyway, so not sure it's worth messing with here

File: scopes/react/ui/docs-app/docs.app-root.tsx
Comment (benjgil) [reply scopes/react/ui/docs-app/docs.app-root.tsx:11]: I'll be refactoring this in a separate PR, where the params will be an object and then we can get rid of the function type, so i'll leave it to there

File: scopes/react/ui/docs-content/docs-content.tsx
Comment (benjgil) [reply scopes/react/ui/docs-content/docs-content.tsx:33]: done (though it still looks pretty much the same - it's just how that syntax looks)

File: scopes/react/ui/properties/properties.tsx
Comment (benjgil) [reply scopes/react/ui/properties/properties.tsx:7]: done

File: scopes/react/ui/properties/properties.tsx
Comment (benjgil) [reply scopes/react/ui/properties/properties.tsx:12]: And i guess that's the point of the hook, you can use it wherever you need to (and it's memo-ised so shouldnt be a loading issue)

File: scopes/react/ui/compositions-app/composition-app.docs.mdx
Comment (benjgil) [reply scopes/react/ui/compositions-app/composition-app.docs.mdx:1]: this is from Uri's changes - the rebase should remove this from the PR

File: scopes/react/ui/apply-providers/apply-providers.tsx
Comment (KutnerUri) [inline scopes/react/ui/apply-providers/apply-providers.tsx:22]: I think you can remove this div. If someone wants to add properties to ApplyProviders, he can just apply a div above it and get the same results.

File: scopes/react/ui/docs-app/docs-app.module.scss
Comment (KutnerUri) [reply scopes/react/ui/docs-app/docs-app.module.scss:21]: sure.

**Review by KutnerUri — APPROVED** [review]

File: scopes/react/ui/properties/properties.tsx
Comment (KutnerUri) [reply scopes/react/ui/properties/properties.tsx]: I guess. Good.

File: scopes/react/react/react.main.runtime.ts
Comment (KutnerUri) [reply scopes/react/react/react.main.runtime.ts:175]: ahh it's was a missing feature in react env. ok 💪

**Review by GiladShoham — APPROVED** [review]

**Pull request merged**
